src/validation/facilityValidation.js

```diff
@@ -0,0 +1,61 @@
+import joi from '@hapi/joi';
+import { validationData } from '../utils';
+
+/**
+ * A collection of validation methods that checks data
+ * entries for Facility as an entity on the App.
+ *
+ * @class FacilityValidator
+ */
+class FacilityValidator {
+  /**
+     * Validates a Facility upon creation
+     *
+     * @param {object} facility - The facility object to be validated.
+     * @returns {object | boolean } - returns an object (error response)
+     * or a boolean if the facility details are valid.
+     */
+  static async validateFacility(facility) {
+    const roomSchema = {
+      occupancyCount: joi.number().required()
+        .label('Please fill in the max number of occupant for this category'),
+      roomCount: joi.number().required()
+        .label('Please fill in the number of rooms that are present for this category'),
+      description: joi.string().min(3).max(250)
+        .label('Please add a valid description'),
+      roomCategoryId: joi.number().required()
+        .label('Please select a category from the options provided'),
+      newCategory: joi.string().min(3).max(10)
+        .label('Please enter a meaningful and short category name')
+    };
+    const schema = {
+      permissionId: joi.number().required()
+        .label('You require some level of access to perform this action'),
+      name: joi.string().min(3).max(25).required()
+        .label('Please enter a valid name for your facility, It should be atleast 3 characters long'),
+      description: joi.string().min(3).max(250).required()
+        .label('Please add a short description'),
+      state: joi.valid(validationData.states).required()
+        .label('Please select a state'),
+      city: joi.string().min(3).max(25).required()
+        .label('Please enter a valid city name'),
+      address: joi.string().min(3).max(100).required()
+        .label('Please fill in a valid Address'),
+      imageUrl: joi.string().uri().required()
+        .label('Please upload an Image of your facility'),
+      amenities: joi.array().items(joi.number()).sparse()
+        .label('Please check the given fields for your choice of amenities'),
+      addOns: joi.array().items(joi.string()).sparse()
+        .label('Please check the given fields for your choice of amenities'),
+      rooms: joi.array().items(joi.object(roomSchema)).required()
+        .label('Please create room categories for your facility')
+    };
+    const { error } = joi.validate({ ...facility }, schema);
+    if (error) {
+      throw error;
```

Comment:
We should stick with throwing the custom error class for the app
```suggestion
      throw new ApiError(errorCode, error.message);
```

Reply:
Thanks, @Segun-Ogundipe for the feedback. The error above is not a custom error, it is handled by the Joi npm package hence, it may not be a good idea to use the custom error class that was created.